addons/httprox/common/src/main/java/org/commonjava/indy/httprox/handler/ProxyResponseWriter.java

```diff
@@ -124,7 +124,7 @@
 
     private final String cls; // short class name for metrics
 
-    private final ThreadPoolExecutor tunnelAndMITMExecutor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
+    private static final ThreadPoolExecutor tunnelAndMITMExecutor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
```

Comment:
I think it would be better for us to pass the Executor in as a parameter, from a component that has access to CDI injection, then use the weft library and @ExecutorConfig to inject a managed executor by name. If we do this, we get more control over thread pool behavior, metrics, and size.

I think it would effectively solve the static / non-static problem, too, since it would use a single Executor instance.

Reply:
Yes this will provide a high level of control. I'll change the commit to your suggested implementation.

Reply:
@jdcasey a question about the implementation. The existing executor uses a cached unbounded pool. You mentioned size which implies an upper bound. Are you suggesting switching to a bounded pool ?